src/vm/builtins/builtin_runner/range_check.zig

```diff
@@ -43,26 +61,188 @@ pub const RangeCheckBuiltinRunner = struct {
         n_parts: u32,
         included: bool,
     ) Self {
+        const bound = Felt252.one().overflowing_shl(16 * n_parts);
+        const _bound: ?Felt252 = if (n_parts != 0 and bound.zero()) null else Felt252.new(bound);
+
         return .{
             .ratio = ratio,
             .base = 0,
             .stop_ptr = null,
             .cell_per_instance = range_check_instance_def.CELLS_PER_RANGE_CHECK,
             .n_input_cells = range_check_instance_def.CELLS_PER_RANGE_CHECK,
-            // TODO: implement shl logic: https://github.com/lambdaclass/cairo-vm/blob/e6171d66a64146acc16d5512766ae91ae044f297/vm/src/vm/runners/builtin_runner/range_check.rs#L48-L53
-            ._bound = null,
+            ._bound = _bound,
             .included = included,
             .n_parts = n_parts,
             .instances_per_component = 1,
         };
     }
 
-    /// Get the base value of this range check runner.
+    /// Get the base value of this Range Check runner.
     ///
     /// # Returns
     ///
     /// The base value as a `usize`.
     pub fn get_base(self: *const Self) usize {
         return self.base;
     }
+
+    /// Get the ratio value of this Range Check runner.
+    ///
+    /// # Returns
+    ///
+    /// The ratio value as an `u32`.
+    pub fn get_ratio(self: *const Self) ?u32 {
```

Comment:
```suggestion
    pub fn getRatio(self: *const Self) ?u32 {
```

> Unlike variables, which are snake_case, functions are camelCase.

source: https://ziglearn.org/chapter-1/#functions

Until the creator accepts the proposal to use snake_case for fns lets stick with camelCase for this (and the rest of your functions too).

Reply:
all of the builtin_runner structs are made with snake case get_base() so was following that formatting. 
https://github.com/keep-starknet-strange/cairo-zig/blob/6c0167b3c8711a5c79fae49f74c161a288531885/src/vm/builtins/builtin_runner/poseidon.zig#L68
Going to assume those are  a mistake. Will reformat.

Reply:
@oxlime Yes, it was probably me who did that, it was indeed a mistake, I had forgotten this subtle change with Rust.